### PR TITLE
Proposal for simplifying adding a dependency group

### PR DESCRIPTION
**Pull request by jmfayard**
To be discussed by phone

### CODE REVIEW

File: plugins/dependencies/src/main/kotlin/dependencies/Kodein.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        val ktor = module("kodein-di-framework-ktor-server-jvm")
+        val tornadofx = module("kodein-di-framework-tornadofx-jvm")
+    }
+}
+
+
+open class DependencyGroup(
+    val group: String,
+    val rule: Rule? = null,
+    val usePlatformConstraints: Boolean = false
```
Comment (LouisCAD) [inline plugins/dependencies/src/main/kotlin/dependencies/Kodein.kt:39]: We'd want to give users the choice, just like we do for Firebase BoM where you can access the `no-BoM` variant.

Comment (jmfayard) [reply plugins/dependencies/src/main/kotlin/dependencies/Kodein.kt:41]: Not a problem, see next commit

File: plugins/dependencies/src/main/kotlin/dependencies/Kodein.kt [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    /**
+     * painless Kotlin dependency injection
+     *
+     * - [Official website here](https://kodein.org/di/)
+     * - GitHub page: [Kodein-Framework/Kodein-DI](https://github.com/Kodein-Framework/Kodein-DI)
+     * - [GitHub Releases here](https://github.com/Kodein-Framework/Kodein-DI/releases)
+     */
+    val di = DI(usePlatformConstraints = true)
+    val `di-no-bom` = DI(usePlatformConstraints = false)
```
Comment (LouisCAD) [inline plugins/dependencies/src/main/kotlin/dependencies/Kodein.kt:18]: This is one of the multiple ways to organise/namespace the no BoM version (and this one is inconsistent with what we have in `Firebase`).

We'll have to discuss this and settle on a convention that we can then mention in the contributing guide.

**Comment by LouisCAD** [comment]
Gives food for thoughts, putting this into my priority list.

**Comment by LouisCAD** [comment]
See issue #275

**Pull request closed without merging**